DEVELOPER_GUIDE.md

```diff
@@ -0,0 +1,155 @@
+# Developer Guide
+
+These instructions are intended to help PureScript users more easily contribute to the compiler, even if this is your first Haskell project.
+
+## Prerequisites
+
+Install `stack`. [Instructions](https://docs.haskellstack.org/en/stable/README/).
+
+See the [PureScript Install Guide](INSTALL.md) for information on the `curses`/`ncurses` dependency.
+
+Update `stack` before proceeding:
+```
+stack update
+```
+
+## Select Release
+
+After cloning the PureScript compiler repo, `checkout` the tagged release that you'd like to build.
+
+For example:
+```
+git checkout v0.13.8
+```
+
+Note that you can view the latest tags with:
+```
+git log --oneline --tags
+```
+
+You may also skip this tagged checkout and build the unreleased `master` branch instead.
+
+Another option is to "unpack" the latest release locally. This creates another directory containing the code for that release, but this code is not tracked by git, so it is less convenient if you are planning on making changes:
+
+```
+stack unpack purescript
+cd purescript-x.y.z  # (replace x.y.z to match the directory just created via unpack)
+```
+
+## Build
+
+```
+stack build
+```
+
+This will take a while the first time it is run.
+
+## Install
+
+If you'd like to test the build in a local shell, you can run:
+```
+stack exec bash
+```
+
+This prepends the location of the compiler to your `$PATH`, so you can test your changes. This works with other tools that depend on the `purs` binary, such as `spago`. Note that rebuilds continue to use this same path, so you don't need to keep re-running `stack exec` to test new changes (assuming you're using a separate terminal for `stack exec`). Remember that `stack exec` must be launched from the same directory that you ran `stack build` from.
+
+You may also install globally with:
+```
+stack install
+```
+
+This copies the compiler executable (`purs`) into `~/.local/bin`.
+
+## Profiling
+
+A profiling build is used to help diagnose performance issues with the compiler.
+
+Create a profiling build with:
+```
+stack build --profile
+```
+This will also take a while the first time it is run. Incremental builds are faster, and you may swap between standard and profiling builds without a rebuild penalty as work for each is saved in separate directories.
```

Comment:
There was some confusion about this behavior in one of the meetings.
So I think it would be helpful to note that these faster rebuilds are available in stack 2.x.

Reply:
Can you be more specific? I don't see how it matters; if you need to do a profiling build, then there's not really any way around that; you just have to wait for as long as it takes.

Reply:
The issue was that build workspaces were shared, so toggling build modes acted like clean rebuilds.
1. Standard build - slow
1. Standard build - fast
1. Profiling build - slow
1. Profiling build - fast
1. Standard build - slow (previous work clobbered by recent profiling build)
1. Profiling build - slow (previous work clobbered by recent standard build)